### PR TITLE
feat(travis): make it possible to specify the list of repos to scan

### PR DESCRIPTION
**Pull request by costimuraru**
Fixes https://github.com/spinnaker/spinnaker/issues/5853

### CODE REVIEW

**Comment by costimuraru** [comment]
CC @jervi

File: igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java [hunk 1]
```diff
@@ -338,6 +346,18 @@ public Build getBuild(String repoSlug, int buildNumber) {
         .collect(Collectors.toList());
   }
 
+  private List<V3Build> getBuildsForSpecificRepos(Collection<String> repositories) {
+    return repositories.stream()
```
Comment (jervi) [inline igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java:350]: Have you tested using a parallellStream here? Don't know how the Travis API will react, but seems like a simple way to speed this up

File: igor-web/config/igor.yml [hunk 2]
```diff
@@ -9,7 +11,7 @@ spinnaker:
     # A lower value can help prevent accidental re-triggers of pipelines, but may require hands-on operations if
     # set low.
     # See code documentation for more details. Class [PollingSafeguardProperties] in
-    # https://github.com/spinnaker/igor/blob/master/igor-web/src/main/groovy/com/netflix/spinnaker/igor/IgorConfigurationProperties.groovy#L47
+    # https://github.com/spinnaker/igor/blob/master/igor-core/src/main/java/com/netflix/spinnaker/igor/IgorConfigurationProperties.java#L92
```
Comment (jervi) [inline igor-web/config/igor.yml:14]: Nice catch

File: igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java (hunk 1 above)
Comment (costimuraru) [reply igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java]: Good idea! I've updated with parallelStream and it looks like it's working quite nicely!

```
2020-06-23 14:04:30.796  INFO 35991 --- [   scheduling-1] c.n.s.igor.travis.service.TravisService  : fetching builds for repo: adobe/ops-cli
2020-06-23 14:04:30.796  INFO 35991 --- [nPool-worker-19] c.n.s.igor.travis.service.TravisService  : fetching builds for repo: adobe/alloy
```

File: igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java [hunk 3]
```diff
@@ -338,6 +346,19 @@ public Build getBuild(String repoSlug, int buildNumber) {
...
+    return repositories
+        .parallelStream()
+        .map(
+            repoSlug -> {
+              log.info("fetching builds for repo: {}", repoSlug);
+              return travisClient.v3builds(
+                  getAccessToken(), repoSlug, buildResultLimit, addLogCompleteIfApplicable());
+            })
+        .flatMap(builds -> builds.getBuilds().stream())
+        .collect(Collectors.toList());
```
Comment (robzienert) [inline igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java:359]: How do exceptions get handled in a `parallelStream`? Will it abort the whole stream (I would assume so). Would you prefer to abort from the entire op, or get a partial response?

Comment (costimuraru) [reply igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java]: Good point, @robzienert. Made the changes to skip the repo if it fails.

File: igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java [hunk 4]
```diff
@@ -338,6 +348,28 @@ public Build getBuild(String repoSlug, int buildNumber) {
...
+    try {
+      log.info("fetching builds for repo: {}", repoSlug);
+      return Either.left(
+          travisClient.v3builds(
+              getAccessToken(), repoSlug, buildResultLimit, addLogCompleteIfApplicable()));
+    } catch (Exception e) {
+      log.error("Failed to fetch builds for repo {}. Skipping. {}.", repoSlug, e.getMessage());
+      return Either.right(e);
+    }
+  }
```
Comment (robzienert) [inline igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java:371]: Nice.

**Pull request merged**